OneCommunityGlobal/HighestGoodNetworkApp
# Shengwei_migrate_TinyMCE

# Description
<img width="782" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/fb1acc3c-5f7f-4380-b570-aee4df48aa8f">

## Related PRS (if any):
N/A

## Main changes explained:
- Remove TinyMCE cloud CDN script from the project
- Remove TinyMCE API key usage from the project
- Update Circle CI config for both PROD and BETA to align with the `development` workflow
- Update `.gitignore` to include `public/tinymce`
- Update scripts in package.json to integrate new script for installing self-hosting TinyMCE
- Upgrade TinyMCE and install required dependencies for self-hosting integration
- Setup a `postinstall.js` script to copy TinyMCE to the public directory for hosting
- Update license_key to 'gpl' and define source file for TinyMCE editors across the application
- Change editor's config from hardcoded string to variable-based string in some files.
- **Update `.eslintignore` file - adding `/public/`, `/build/`, and `/node_modules/`.** Please comment if you have any questions.
- Include license info in `package.json` - we need to comply GLPv2 for utilizing self-hosting TinyMCE. **Please leave comments on this too if you have any concerns about the software licensing!**
- Update `README.md` to include new installation step and software license agreement

## How to test:
1. check into current branch
2. do `npm install` and `npm run postinstall` to run this PR locally
3. Verify a `tinymce` folder created under the `public` directory
4. Clear site data/cache
5. login as admin user
6. Verify every editor is functional and does not show a warning about the API key is required. (e.g. summary submission, badge development, or any component that use an editor)
<img width="460" alt="image" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/69882989/3f8a5e6b-95d9-444e-a933-7c97ee4df35b">

7.  Do `npm run build` and verify that the production build application is functional


## Screenshots or videos of changes:

## Note:
Include the information the reviewers need to know.
